Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs

```diff
@@ -57,45 +38,72 @@ public void SaveGraphicPlayerPrefs()
 		public void LoadGraphicPlayerPrefs()
 		{
 			
-			var loaded = JsonUtility.FromJson<GeneralGraphics.Data>(PlayerPrefs.GetString("GraphicsData"));
-				
+			var loaded = JsonUtility.FromJson<GeneralGraphics.GraphicsData>(PlayerPrefs.GetString("GraphicsData"));
+
+			GeneralGraphic.CustomSettings.isOn = loaded.CustomSettings;
 			GeneralGraphic.TextureQuality.value = loaded.TextureQuality;
 			GeneralGraphic.ShadowRes.value = loaded.ShadowRes;
 			GeneralGraphic.AntiAliasing.value = loaded.AntiAliasing;
 			GeneralGraphic.Shadows.value = loaded.Shadows;
 			GeneralGraphic.VSync.isOn = loaded.VSync;
 			GeneralGraphic.SoftParticles.isOn = loaded.SoftParticles;
-			GeneralGraphic.CustomSettings.isOn = loaded.CustomSettings;
+			
 
 			GeneralGraphic.UpdateEverything();
 
-			var _loaded = JsonUtility.FromJson<QualitySwitcher.Data>(PlayerPrefs.GetString("QualityProfile"));
+			var _loaded = JsonUtility.FromJson<QualitySwitcher.QualityData>(PlayerPrefs.GetString("QualityProfile"));
 			QualitySwitcher.Slider.value = _loaded.Slider;
 			
 			label.color = Color.green;
 			label.text = "loaded player prefs";
+
+			if(loaded.CustomSettings)
+			{
+				if(QualitySettings.GetQualityLevel() != 6)
+				{
+					QualitySettings.SetQualityLevel(6, true);
+				}
+			}
+
+			GeneralGraphic.UpdateEverything();
+			//QualitySwitcher.UpdateQuality();
 		}
 
 		public void AdvancedOptions()
 		{
 			var selected = CustomSettings.isOn;
-			var temp = QualitySettings.GetQualityLevel();
 			
 			if(!selected)
 			{
-				QualitySettings.SetQualityLevel(temp);
+				GeneralGraphic.TextureQuality.interactable = false;
```

Comment:
These six `GeneralGraphic.TextureQuality.interactable = false` can also just be replaced with `GeneralGraphic.TextureQuality.interactable = selected`, so if we aren't selected, we set them to false, otherwise they become true.